tests/NgTests/SearchIndexFromCatalogCollectorTests.cs

```diff
@@ -102,6 +111,63 @@ public async Task DoesNotSkipPackagesWhenExceptionOccurs(string catalogUri, stri
 
                     Assert.Equal(DateTime.Parse(expectedCursorBeforeRetry).ToUniversalTime(), cursorBeforeRetry);
                     Assert.Equal(DateTime.Parse("2015-10-12T10:08:55.3335317Z").ToUniversalTime(), cursorAfterRetry);
+
+                    telemetryService.Verify(t => t.TrackIndexCommitDuration(), Times.Exactly(2));
+                    telemetryService.Verify(t => t.TrackIndexCommitTimeout(), Times.Never);
+                    indexCommitDurationMetric.Verify(m => m.Dispose(), Times.Exactly(2));
+                }
+            }
+        }
+
+        [Fact]
+        public async Task ThrowsIfCommitTimesOut()
+        {
+            // Arrange
+            var storage = new MemoryStorage();
+            var storageFactory = new TestStorageFactory(name => storage.WithName(name));
+
+            MockServerHttpClientHandler mockServer;
+            mockServer = new MockServerHttpClientHandler();
+            mockServer.SetAction("/", request => Task.FromResult(new HttpResponseMessage(HttpStatusCode.OK)));
+
+            var catalogStorage = Catalogs.CreateTestCatalogWithOnePackage();
+            await mockServer.AddStorageAsync(catalogStorage);
+
+            ReadWriteCursor front = new DurableCursor(
+                storage.ResolveUri("cursor.json"),
+                storage,
+                MemoryCursor.MinValue);
+            ReadCursor back = MemoryCursor.CreateMax();
+
+            var commitTimeout = TimeSpan.FromSeconds(1);
+            var stuckDuration = TimeSpan.FromMinutes(1);
```

Comment:
Adds a minute to tests, or are they run in parallel? Could this be shorter?

Reply:
The timeout is set to a second, so this test completes after a second.